bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/protocol/MPDConnection.java

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mpd.internal.protocol;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.mpd.internal.handler.MPDEventListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class for communicating with the music player daemon through a IP connection
+ *
+ * @author Stefan Röllin - Initial contribution
+ */
+@NonNullByDefault
+public class MPDConnection implements MPDResponseListener {
+
+    private static final int DISPOSE_TIMEOUT_MS = 1000;
+
+    private final Logger logger = LoggerFactory.getLogger(MPDConnection.class);
+
+    private @Nullable MPDConnectionThread connectionThread = null;
+    private MPDEventListener listener;
+
+    /**
+     * Constructor
+     *
+     * @param address the IP address of the music player daemon
+     * @param port the TCP port to be used
+     * @param password the password to connect to the music player daemon
+     */
+    public MPDConnection(MPDEventListener listener) {
+        this.listener = listener;
+    }
+
+    /**
+     * start the connection
+     *
+     * @param address the IP address of the music player daemon
+     * @param port the TCP port to be used
+     * @param password the password to connect to the music player daemon
+     */
+    public void start(String address, Integer port, String password) {
+        if (connectionThread == null) {
+            connectionThread = new MPDConnectionThread(this, address, port, password);
+            connectionThread.start();
+        }
+    }
+
+    /**
+     * dispose the connection
+     */
+    public void dispose() {
+        final MPDConnectionThread connectionThread = this.connectionThread;
+        if (connectionThread != null) {
+            connectionThread.dispose();
+            connectionThread.interrupt();
+            try {
+                connectionThread.join(DISPOSE_TIMEOUT_MS);
+            } catch (InterruptedException e) {
```

Comment:
Consider renaming to `ignore`
```suggestion
            } catch (InterruptedException ignore) {
```